### DISCUSSION TITLE
Getting started...

### DISCUSSION THREAD

**Discussion by amtrak115** [perpetualKid/FreeTrainSimulator#65] [General]
do you just run build.cmd from command line or withing Visual Studio??  Sorry for dumb questions   I'm trying to get up to speed here.

**Comment by perpetualKid** [comment]
typically using Visual Studio (2019) and the solution file Source/Orts.sln. If you open first time, there are couple nuget packages which will be restored, so may take a minute or two.
The build.cmd are remains from the automated build process of the original OR version and should likely be removed, I'll add a task for it.

For now development should be based on the https://github.com/perpetualKid/ORTS-MG/tree/Monogame branch - master is mainly used for merging efforts with ORNYMG and orignal OR, but does not always contains the latest development changes. 

While the new TrackViewer work was originally born in the Dispatcher-Monogame branch (https://github.com/perpetualKid/ORTS-MG/tree/Dispatcher-Monogame), the latest version had been merged into Monogame branch already, so it's safe to just use Monogame branch as baseline.

**Comment by perpetualKid** [comment]
@amtrak115 If you prefer command line, just switch to the \Source directory, and run
`dotnet restore` 
followed by
`dotnet build` 

either way, the build output will be in \Program. There you'll see two folders, .Net48 and .netcoreapp3.1. Both basically contain the same version, just one is build for .NET 4.8 (which we plan to drop soon) and the other for .NET Core 3.1. This is purely for backward compatiblity for existing users, but I'd suggest to only focus on .NET Core development. Given the dependency on .NET 4.8 still, you can use C# language version 7.3 features, this will change only once we got rid of .NET FX.

Maybe worth we put some documentation about getting started in the wiki :)

**Reply by amtrak115** [reply]
thanks for the info...will give it a try
